<launch>
    <arg name="self_id" default="0" />
    <arg name="is_4dof" default="false" />
    <arg name="solver_type" default="arock" />
    <arg name="g2o_path" default="" />
    <arg name="output_path" default="" />
    <arg name="wait_for_start" default="false" />
    <arg name="ceres_max_solver_time" default="100.0" />
    <arg name="ceres_max_num_iterations" default="1000" />
    <arg name="max_steps" default="10" />
    <arg name="rho_frame_T" default="0.006" />
    <arg name="rho_frame_theta" default="0.02" />
    <arg name="rho_rot_mat" default="0.1" />
    <arg name="eta_k" default="0.9" />
    <arg name="enable_rot_init" default="true" />
    <arg name="rot_init_enable_float32" default="true" />
    <arg name="rot_init_enable_gravity_prior" default="false" />
    <arg name="debug_rot_init_only" default="false" />
    <arg name="enable_linear_pose6d_solver" default="false" />
    <arg name="linear_pose6d_iterations" default="2" />
    <arg name="is_multi" default="false" />
    <arg name="drone_num" default="1" />
    <arg name="ignore_infor" default="false" />
    <arg name="simulate_delay_ms" default="0.0" />
    <arg name="max_solving_time" default="10.0" />
    <arg name="rot_init_state_eps" default="0.1" />

    <node name="d2pgo_test_$(arg self_id)" pkg="d2pgo" type="d2pgo_test" output="screen" >
        <param name="self_id" value="$(arg self_id)" type="int" />
        <param name="is_4dof" value="$(arg is_4dof)" type="bool" />
        <param name="solver_type" value="$(arg solver_type)" type="string" />
        <param name="g2o_path" value="$(arg g2o_path)" type="string" />
        <param name="output_path" value="$(arg output_path)" type="string" />
        <param name="wait_for_start" value="$(arg wait_for_start)" type="bool" />
        <param name="ceres_max_solver_time" value="$(arg ceres_max_solver_time)" type="double" />
        <param name="ceres_max_num_iterations" value="$(arg ceres_max_num_iterations)" type="int" />
        <param name="max_steps" value="$(arg max_steps)" type="int" />
        <param name="rho_frame_theta" value="$(arg rho_frame_theta)" type="double" />
        <param name="rho_frame_T" value="$(arg rho_frame_T)" type="double" />
        <param name="rho_rot_mat" value="$(arg rho_rot_mat)" type="double" />
        <param name="eta_k" value="$(arg eta_k)" type="double" />
        <param name="enable_rot_init" value="$(arg enable_rot_init)" type="bool" />
        <param name="rot_init_enable_gravity_prior" value="$(arg rot_init_enable_gravity_prior)" type="bool" />
        <param name="rot_init_enable_float32" value="$(arg rot_init_enable_float32)" type="bool" />
        <param name="debug_rot_init_only" value="$(arg debug_rot_init_only)" type="bool" />
        <param name="enable_linear_pose6d_solver" value="$(arg enable_linear_pose6d_solver)" type="bool" />
        <param name="linear_pose6d_iterations" value="$(arg linear_pose6d_iterations)" type="int" />
        <param name="is_multi" value="$(arg is_multi)" type="bool" />
        <param name="drone_num" value="$(arg drone_num)" type="int" />
        <param name="ignore_infor" value="$(arg ignore_infor)" type="bool" />
        <param name="simulate_delay_ms" value="$(arg simulate_delay_ms)" type="double" />
        <param name="max_solving_time" value="$(arg max_solving_time)" type="double" />
        <param name="rot_init_state_eps" value="$(arg rot_init_state_eps)" type="double" />
    </node>
</launch>